### ISSUE TITLE
subscribe does not work if the rpc server requires username & password

### ISSUE THREAD

**Issue by blouflashdb** [onmax/albatross-rpc-client-ts#54]
```
const subscription = await client.blockchainStreams.subscribeForBlocks();
subscription.next((data) => console.log(data));
```
always returns

```
{
  data: undefined,
  metadata: undefined,
  error: { code: 1000, message: "{}" }
}
```

**Comment by blouflashdb** [comment]
It seems like username and password is not supported in the WebSocketClient

```
constructor(url: URL | string, auth?: Auth) {
    if (typeof url === 'string') {
      url = new URL(url)
    }
    const wsUrl = new URL(url.href.replace(/^http/, 'ws'))
    wsUrl.pathname = '/ws'
    this.url = wsUrl
    this.textDecoder = new TextDecoder()

    if (auth && 'secret' in auth && auth.secret) {
      this.url = new URL(`${url.href}?secret=${auth.secret}`)
    }
  }
```

**Comment by onmax** [comment]
fixed nimiq-rpc-client-ts@0.5.0
